### PR TITLE
0.18.1

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by mdeabreu** [comment]
I think there are a couple of things going on here.

First, the latest release of Lemmy (0.18.1) requires Postgres 15, this should be satisfied by Debian 12 Bookworm once YunoHost upgrades to add support for it.

Second, the error seen in testing seems to be as a result of the `sharp` node package. This could probably be fixed with the following `ynh_npm install --global sharp` (not 100% sure if `--global` is necessary).

I spun up a VM with a clean install of Debian 12, installed YunoHost using the bookworm install script, added the npm install line to the Lemmy install script and everything seems to be working as expect.

**Comment by ericgaspar** [comment]
!testme

**Comment by mdeabreu** [comment]
Just ran the latest set of changes up to `808495b` and unfortunately it still breaks, I did some local testing and it looks like it can be fixed quite easily. Simply change these two lines

```
#=================================================
# INSTALL THE SHARP
#=================================================

- pushd $install_dir
+ pushd $install_dir/lemmy-ui
 	ynh_use_nodejs
- 	ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH yarn global add sharp
+ 	ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH yarn add sharp
popd
```

This will change our directory to the `lemmy-ui` subdirectory which is the part of the install that's failing. `lemmy-ui` already comes with `sharp` but it is built for the docker container so we will install it ourself over top of the copy of sharp they have (remove `global`).

As an added bonus this seems to work on Debian 11!

File: scripts/install [hunk 1]
```diff
@@ -47,6 +46,15 @@ ynh_script_progression --message="Configuring NGINX web server..." --weight=1
 # Create a dedicated NGINX config
 ynh_add_nginx_config
 
+#=================================================
+# INSTALL THE SHARP
+#=================================================
+
+pushd $install_dir
```
Comment (mdeabreu) [inline scripts/install:53]:
```suggestion
pushd $install_dir/lemmy-ui
```

File: scripts/install [hunk 2]
```diff
@@ -47,6 +46,15 @@ ynh_script_progression --message="Configuring NGINX web server..." --weight=1
...
+ 	ynh_use_nodejs
+	ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH yarn global add sharp
```
Comment (mdeabreu) [inline scripts/install:55]:
```suggestion
	ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH yarn add sharp
```

Comment (mdeabreu) [inline scripts/install:55]:
```suggestion
pushd $install_dir/lemmy-ui
 	ynh_use_nodejs
	ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH yarn add sharp
```

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
@mdeabreu thank you for your help! merging to testing

**Pull request merged**